apps-rendering/src/components/LiveBlocks/index.tsx

```diff
@@ -30,12 +32,29 @@ const contributorToBlockContributor = (
 interface LiveBlocksProps {
 	blocks: LiveBlock[];
 	format: ArticleFormat;
+	pinnedPost: Option<LiveBlock>;
+	pageNumber: number;
 }
 
-const LiveBlocks: FC<LiveBlocksProps> = ({ blocks, format }) => {
+const LiveBlocks: FC<LiveBlocksProps> = ({
+	blocks,
+	format,
+	pinnedPost,
+	pageNumber,
+}) => {
+	const pinnedPostId = compose(
+		withDefault(''),
+		map((pinnedPost: LiveBlock) => pinnedPost.id),
+	)(pinnedPost);
+
+	const showPinnedPost =
+		pageNumber === 1 && pinnedPost.kind === OptionKind.Some;
+
 	return (
 		<>
 			{/* Accordion? */}
+			{/* below is placeholder for pinned post component */}
+			{showPinnedPost && <></>}
```

Comment:
Out of curiosity, what made you choose to render this inside the `LiveBlocks` component instead of the `LiveLayout` component?

Reply:
It'd be fine to render it in the `LiveLayout` component, I just thought it might be cleaner to move it into the LiveBlocks, because when the `PinnedPost` component is created in future PR, we need to pass `LiveBlockContainer` as children to it, and that could make the `LiveLayout` a bit messy (AR `LiveLayout` is currently very easy to follow and a lot simpler than the DCR one :) )


ALSO, now that I removed the `pinnedPost` field from `Item` and am deriving it within the `LiveBlocks` component, I think it's ever more important that we render the `PinnedPost` component within `LiveBlocks`

Reply:
That all makes sense, thanks for explaining!